client/my-sites/plugins/plugin-management-v2/plugin-action-status/render-status-message.tsx

```diff
@@ -0,0 +1,180 @@
+import classNames from 'classnames';
+import { translate } from 'i18n-calypso';
+import { ReactElement, ReactNode } from 'react';
+import {
+	ACTIVATE_PLUGIN,
+	DEACTIVATE_PLUGIN,
+	DISABLE_AUTOUPDATE_PLUGIN,
+	ENABLE_AUTOUPDATE_PLUGIN,
+	REMOVE_PLUGIN,
+} from 'calypso/lib/plugins/constants';
+import {
+	PLUGIN_INSTALLATION_COMPLETED,
+	PLUGIN_INSTALLATION_ERROR,
+	PLUGIN_INSTALLATION_IN_PROGRESS,
+} from 'calypso/state/plugins/installed/status/constants';
+import { CurrentSiteStatus, PluginActionStatus, PluginActionStatusMessage } from '../types';
+import type { SiteDetails } from '@automattic/data-stores';
+
+import './style.scss';
+
+interface Props {
+	selectedSite?: SiteDetails;
+	groupKey: PluginActionStatus;
+	groupedStatues: { [ key in PluginActionStatus ]: Array< CurrentSiteStatus > };
+	hasOneStatus: boolean;
+}
+
+const getStatusMessage = (
+	count: number,
+	hasOneStatus: boolean,
+	selectedSite?: SiteDetails
+): PluginActionStatusMessage => {
+	const translationArgs = {
+		args: { count },
+		count,
+	};
+
+	const failedStatusMessage = ( message: ReactNode ) =>
+		hasOneStatus
+			? message
+			: translate( 'Failed on %(count)s site', 'Failed on %(count)s sites', translationArgs );
```

Comment:
Do we need this general purpose message because we don't have enough space when there are multiple statuses?

Reply:
Yes, we are using this generic failure message where there are multiple statuses; you can see the screenshot of how we show multiple statuses.